File: 2018/test/day16.test.ts
import fs from "fs";
import assert from "assert";
import { part1, part2 } from "../src/day16";

describe("Day 16", function () {
  it("Part 1 - Sample", function () {
    const input = readInput("input/day16-sample.txt");
    assert.equal(part1(input), 1);
  });

  it("Part 1", function () {
    const input = readInput("input/day16-part1.txt");
    assert.equal(part1(input), 529);
  });

  it("Part 2", function () {
    const sampleInput = readInput("input/day16-part1.txt");
    const program = readInput("input/day16-part2.txt");
    assert.equal(part2(sampleInput, program), 573);
  });
});

function readInput(path: string) {
  return fs.readFileSync(path).toString();
}

File: 2022/src/test/kotlin/Day16Test.kt
import org.junit.Assert.*
import org.junit.Test
import utils.getResourceAsString

class Day16Test {

  @Test
  fun part1_sample() {
    assertEquals(1651, Day16.part1(getResourceAsString("day16-sample.txt")))
  }

  @Test
  fun part1() {
    assertEquals(2320, Day16.part1(getResourceAsString("day16.txt")))
  }

  @Test
  fun part2_sample() {
    assertEquals(1707, Day16.part2(getResourceAsString("day16-sample.txt")))
  }

  @Test
  fun part2() {
    assertEquals(2967, Day16.part2(getResourceAsString("day16.txt")))
  }

}